hydrolib/core/utils.py

```diff
@@ -16,6 +17,37 @@ def example(a: float, b: float = 1.0) -> float:
 
 
 def to_key(string: str) -> str:
+    """
+    Construct a key name from a given field name.
+    The given field name may be a Pydantic Field alias, and the key name
+    is intended to be used as a BaseModel class member variable name.
+
+    Args:
+        string (str): input field name
+
+    """
+    # First replace any leading digits, because those are undesirable
```

Comment:
Maybe replace all digits? Bit inconsistent to have oned2dfield? Would simplify the code underneath as well, removing all `re` parts. Or the more extreme and ugly alternative in the other direction, just replace 1 if it occurs, this probably only happens on the `1d2dlinkfile` field and no other fields contain a single `1`?

Reply:
I did this originally, but changed plans, because a lot of class members then changed, such as:
`nonlin1d`, `teta0`, `uniffrictcoef1d`, etc., etc.
The trick is dirty enough by itself and only intended to avoid invalid Python variable names starting with digits. Beyond that, I don't want to fiddle to much with our model class fields.